SpatialGDK/Source/Private/Interop/SpatialView.cpp

```diff
@@ -125,50 +125,79 @@ Worker_Authority USpatialView::GetAuthority(Worker_EntityId EntityId, Worker_Com
 	return WORKER_AUTHORITY_NOT_AUTHORITATIVE;
 }
 
-improbable::UnrealMetadata* USpatialView::GetUnrealMetadata(Worker_EntityId EntityId)
+template <typename T>
+typename T* USpatialView::GetComponentData(Worker_EntityId EntityId)
 {
-	if (TSharedPtr<improbable::UnrealMetadata>* UnrealMetadataPtr = EntityUnrealMetadataMap.Find(EntityId))
+	if (TMap<Worker_ComponentId, TUniquePtr<ComponentStorageBase>>* ComponentStorageMap = EntityComponentMap.Find(EntityId))
 	{
-		return UnrealMetadataPtr->Get();
-	}
-
-	return nullptr;
-}
-
-improbable::EntityAcl* USpatialView::GetEntityACL(Worker_EntityId EntityId)
-{
-	if (TSharedPtr<improbable::EntityAcl>* EntityAclPtr = EntityACLMap.Find(EntityId))
-	{
-		return EntityAclPtr->Get();
+		if (TUniquePtr<improbable::ComponentStorageBase>* Component = ComponentStorageMap->Find(T::ComponentId))
+		{
+			return &(static_cast<improbable::ComponentStorage<T>*>(Component->Get())->Get());
```

Comment:
I'm pretty sure we should return weak pointers here instead of returning a raw pointer to a unique ptr. 

https://isocpp.org/blog/2013/07/quick-q-whats-the-weak-ptr-equivalent-for-unique-ptr-stackoverflow